### ISSUE TITLE
The specified item could not be found in the keychain

### ISSUE THREAD

**Issue by klvnchua** [duraki/SketchCrapp#62]
**Describe the issue [required]**
I'm getting an error of `error: The specified item could not be found in the keychain.`

If you can point me to the right direction that'll be greatly appreciated!! Thanks so much

**macOS version [required]**
macOS BigSur 11.3 beta 2

**Sketch version [required]**
v70.5

**Terminal error message [required]**
Here's the full output

```
           __       __      __
      ___ / /_____ / /_____/ /  ___________ ____  ___
    ( _-</  '_/ -_) __/ __/ _ \/ __/ __/ _ `/ _ \/ _ \
    /___/_/\_\\__/\__/\__/_//_/\__/_/  \_,_/ .__/ .__/
                                          /_/  /_/
         Sketch.App Patch Tool (https://github.com/duraki/SketchCrapp)
         by @duraki & @elijahtsai

[+] Checking directory tmp existence ... OK
[+] Fetching https://download.sketch.com/sketch-70.5-109233.zip ...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 58.5M  100 58.5M    0     0  17.6M      0  0:00:03  0:00:03 --:--:-- 17.6M
[+] Checking if Sketch.app exist in /tmp ... Not exist. Continuous.
[+] Checking if Sketch.app exist in /Applications ... Exist. Removing.
[+] Moving Sketch.app to /Applications directory ... Successfully.
[+] Analysing application bundle ... Starting
[+] Finding executable file ... OK
[+] Finding Info.plist ... OK
[+] Checking Info.plist for CFBundleShortVersionString ... OK
[+] Validating executable file ... OK
[+] Selected Sketch.app version is 70.5 ... SketchCrapp starting ... OK
[+] Patching offsets for 70.5 ... Starting modern arch patch via bash&seek ...
[+] Patching address at offset: 0x585eef with value: \00
1+0 records in
1+0 records out
1 bytes transferred in 0.000015 secs (67650 bytes/sec)
[+] Patching address at offset: 0x585ef2 with value: \00
1+0 records in
1+0 records out
1 bytes transferred in 0.000007 secs (139810 bytes/sec)
[+] Patching address at offset: 0x584b3e with value: \00\00
2+0 records in
2+0 records out
2 bytes transferred in 0.000010 secs (199729 bytes/sec)
[+] Patching address at offset: 0x584c6e with value: \165
1+0 records in
1+0 records out
1 bytes transferred in 0.000009 secs (110376 bytes/sec)
[+] Patching address at offset: 0x66ce63 with value: \00\00\00\00\00\00\00\00\00\00\00\00\00\00\00
15+0 records in
15+0 records out
15 bytes transferred in 0.000030 secs (499322 bytes/sec)
[+] Patching address at offset: 0x66ce73 with value: \40\123\153\145\164\143\150\103\162\141\160\160\40
13+0 records in
13+0 records out
13 bytes transferred in 0.000026 secs (500238 bytes/sec)
[+] Patching address at offset: 0xdcd404 with value: \01
1+0 records in
1+0 records out
1 bytes transferred in 0.000006 secs (161319 bytes/sec)
[+] Patching address at offset: 0xdcd408 with value: \24
1+0 records in
1+0 records out
1 bytes transferred in 0.000008 secs (123362 bytes/sec)
[+] Patching address at offset: 0xdcc07c with value: \165\00
2+0 records in
2+0 records out
2 bytes transferred in 0.000006 secs (335544 bytes/sec)
[+] Patching address at offset: 0xdcc19f with value: \64
1+0 records in
1+0 records out
1 bytes transferred in 0.000007 secs (144631 bytes/sec)
[+] Patching address at offset: 0xed0fbc with value: \00\00\00\00\00\00\00\00\00\00\00\00\00\00\00
15+0 records in
15+0 records out
15 bytes transferred in 0.000033 secs (455903 bytes/sec)
[+] Patching address at offset: 0xed0fcc with value: \40\123\153\145\164\143\150\103\162\141\160\160\40
13+0 records in
13+0 records out
13 bytes transferred in 0.000024 secs (539861 bytes/sec)
[+] Checking user default keychain ... Exist.
[+] SketchCrapp certificate already exists.
[+] Skipping certificate creation ... OK
[+] Signing the patched *.app bundle. This may require root privilege.
[+] If asked, enter your login password. Choose "Always Allow" to not be asked again.
error: The specified item could not be found in the keychain.
[-] Failed to Signing Sketch bundle.
[+] Cleaning up file(s) ... Cleaned
[+] SketchCrapp last published date: 2021-02-17 serial 001
```

**Additional context [optional]**
M1 Mac

**Screenshots [optional]** 
Add screenshots to help explain your problem.

**Comment by elijahtsai** [comment]
Hi @klvnchua , Are you using other keychain other then “login” and “System”? Things like your personal keychain... etc. If you have, try to unlock it or unplugged it if keychain is in the external drive. Or you can try to restart your mac to see if those step are help if not get back to me , I’ll see if there have other methods can help you.

**Comment by elijahtsai** [comment]
Also, are you ever using early version of version 70, like 70.4 or 70.3?

**Comment by elijahtsai** [comment]
If this issue does not get any reply before 2021-02-28 00:00:00 will be treat as Not Important and close by us. if you still having a question, you may need to open another one.

**Comment by klvnchua** [comment]
Hi there apologies for the late response, had a crazy week. Yup I don't use anything else other than the MacOS keychain. I've tried using `-m` as well no luck :(

**Comment by elijahtsai** [comment]
@klvnchua 
Next I’ll suggest you to delete our certificate and let it re-generate again, open the keychain app, and at the top-right corner search “sketchcrapp”, in the result list delete the file match identical with “sketchcrapp” (normally it will have two files one for certificate one for private key), and run the script again see if this help

**Comment by klvnchua** [comment]
@elijahtsai Thanks for getting back to me, no luck with this solution as well.

**Comment by elijahtsai** [comment]
🤨
Because I can’t recreate the same error you met, also I and mr.duraki don’t have M1 computer so we don’t know if this problem only in M1 , I’ll leave this issue **open** until have a **reasonable answer** and let other user report if they met same issue as you are. 
The right way I can now give you is the error in your log `error: The specified item could not be found in the keychain.` this is cause by codesign command, but I can’t found any solution that can solve your problem  , also we update to support 70.6 try to re-download script again see if this solve your problem magically 😂

**Comment by elijahtsai** [comment]
Hi @klvnchua I've update the signing process, can you try again and let me known if it worked for you.
Don't forget to re-download our script again from GitHub. The latest publish date is 2021-03-07 serial 005

**Comment by klvnchua** [comment]
Hi @elijahtsai thanks so much for working on this, I ended up switching back to my intel Mac so its been working well so far (amazing work guys!). I've returned my M1 mac to Apple due to a few other issues as well so I can't test it anymore :( apologies.

**Comment by elijahtsai** [comment]
It’s fine, my friend, good to hear the news, ok i’ll close this because I think this should fixed the problem. Wish you have wonderful day. ☺️
Closed #62
